### ISSUE TITLE
Multiple files support

### ISSUE THREAD

**Issue by huynhquancam** [sindresorhus/gulp-ruby-sass#253]
Hi guys, will there be a support for multiple files?

```js
var rubySass = require("gulp-ruby-sass");

gulp.task("sass", function() {
  rubySass(["webapp/src/main.scss", "backoffice/src/main.scss"], { // options omitted });
});

**Comment by robwierzbowski** [comment]
Does gulp.src support that syntax?

On Sunday, August 30, 2015, Cam Huynh (Jack) <notifications@github.com>
wrote:

> Hi guys, will there be a support for multiple files?
>
> var rubySass = require("gulp-ruby-sass");
>
> gulp.task("sass", function() {
>   rubySass(["webapp/src/main.scss", "backoffice/src/main.scss"], { // options omitted });
> });
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sindresorhus/gulp-ruby-sass/issues/253>.
>


-- 
Rob Wierzbowski
@robwierzbowski <http://twitter.com/#!/robwierzbowski>
http://github.com/robwierzbowski
http://robwierzbowski.com

**Comment by huynhquancam** [comment]
@robwierzbowski Yes, I guess it does.

https://github.com/gulpjs/gulp/blob/master/docs/API.md#gulpsrcglobs-options

**Comment by robwierzbowski** [comment]
Then yes! I'll add it to my list for 2.0!

On Sunday, August 30, 2015, Cam Huynh (Jack) <notifications@github.com>
wrote:

> @robwierzbowski <https://github.com/robwierzbowski> Yes, I guess it does.
>
> https://github.com/gulpjs/gulp/blob/master/docs/API.md#gulpsrcglobs-options
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sindresorhus/gulp-ruby-sass/issues/253#issuecomment-136154586>
> .
>


-- 
Rob Wierzbowski
@robwierzbowski <http://twitter.com/#!/robwierzbowski>
http://github.com/robwierzbowski
http://robwierzbowski.com

**Comment by huynhquancam** [comment]
Awesome :+1:

**Comment by robwierzbowski** [comment]
Oh, by the way you can currently compile multiple files with a directory source, and soon with a glob source (#251). Doesn't look like that would work with your directory structure, but wanted to point out it's possible.

**Comment by robwierzbowski** [comment]
With your above task, what would you expect the dest folder output to be?

I'm thinking of a couple different ways to do this, but because we have to rewrite vinyl file paths and sourcemap paths to get gulp and Sass playing together as efficiently as possible, I'm seeing some potential issues. Maybe not issues, but just more computation on our part.

With `["/foo/bar/webapp.scss", "foo/bar/backoffice.scss"]` we'll want to look at both and make the base path (what gets stripped off when output to `gulp.dest`) the directories they share (`foo/bar`). I suppose this carries over to glob paths too. `["foo/baz/**/*.scss", "foo/bar/(x|y)/*.scss"]` would have `foo` as a base directory.

Thoughts?

**Comment by robwierzbowski** [comment]
Playing around with this, looks like it's going to work.

**Comment by robwierzbowski** [comment]
@huynhquancam, would you like to test #257?

**Comment by huynhquancam** [comment]
@robwierzbowski Sure, I will let you know the result. Thanks for your help.

**Comment by robwierzbowski** [comment]
Fixed with #257!
